packages/market-service/src/idle/idle.ts

```diff
@@ -0,0 +1,89 @@
+import { fromAssetId, toAssetId } from '@shapeshiftoss/caip'
+import { ethereum } from '@shapeshiftoss/chain-adapters'
+import { IdleInvestor } from '@shapeshiftoss/investor-idle'
+import { MarketCapResult, MarketData, MarketDataArgs } from '@shapeshiftoss/types'
+import * as unchained from '@shapeshiftoss/unchained-client'
+
+import { MarketService } from '../api'
+import { ProviderUrls } from '../market-service-manager'
+
+export class IdleMarketService implements MarketService {
+  baseUrl = ''
+  providerUrls: ProviderUrls
+  idleInvestor: IdleInvestor
+
+  constructor({ providerUrls }: { providerUrls: ProviderUrls }) {
+    this.providerUrls = providerUrls
+    this.idleInvestor = new IdleInvestor({
+      chainAdapter: new ethereum.ChainAdapter({
+        providers: {
+          ws: new unchained.ws.Client<unchained.ethereum.Tx>(
+            this.providerUrls.unchainedEthereumWsUrl,
+          ),
+          http: new unchained.ethereum.V1Api(
+            new unchained.ethereum.Configuration({
+              basePath: this.providerUrls.unchainedEthereumHttpUrl,
+            }),
+          ),
+        },
+        rpcUrl: this.providerUrls.jsonRpcProviderUrl,
+      }),
+      providerUrl: this.providerUrls.jsonRpcProviderUrl,
+    })
+    ;(async () => {
```

Comment:
Since we do this yeefee in a constructor but constructors are synchronous, it's not going to be blocking the way we think this is, so this is unsafe - hence the re-initialization in other methods